native/c/zds.cpp

```diff
@@ -606,6 +607,95 @@ typedef struct
 #define BUFF_SIZE 1024
 #define FIELD_LEN 8
 
+void load_dsorg_from_dscb(DSCBFormat1 *dscb, string *dsorg)
+{
+  // Bitmasks translated from binary to hex from "DFSMSdfp advanced services" PDF, Chapter 1 page 7 (PDF page 39)
+  *dsorg = "";
+
+  // PS: 0100 000x ...
+  if (((dscb->ds1dsorg >> 8) & 0xF0) == 0x40)
+  {
+    *dsorg = ZDS_DSORG_PS;
+  }
+  // DA: 0010 000x ...
+  else if (((dscb->ds1dsorg >> 8) & 0xF0) == 0x20)
+  {
+    *dsorg = ZDS_DSORG_DA;
+  }
+  // PO: 0000 001x ...
+  else if (((dscb->ds1dsorg >> 8) & 0x0E) == 0x2)
+  {
+    *dsorg = ZDS_DSORG_PO;
+  }
+  // VSAM: ... 000x 10xx
+  else if ((dscb->ds1dsorg & 0x0C) == 0x8)
+  {
+    *dsorg = ZDS_DSORG_VSAM;
+  }
+
+  // Unmovable: Last bit of first half is set
+  if (((dscb->ds1dsorg >> 8) & 0x01) == 0x1)
+  {
+    *dsorg += 'U';
+  }
```

Comment:
Thoughts on updating the bitmasks to avoid the bit shift?


```suggestion
  // PS: 0100 000x ...
  if ((dscb->ds1dsorg & 0xF000) == 0x4000)
  {
    *dsorg = ZDS_DSORG_PS;
  }
  // DA: 0010 000x ...
  else if ((dscb->ds1dsorg & 0xF000) == 0x2000)
  {
    *dsorg = ZDS_DSORG_DA;
  }
  // PO: 0000 001x ...
  else if ((dscb->ds1dsorg & 0x0E00) == 0x0200)
  {
    *dsorg = ZDS_DSORG_PO;
  }
  // VSAM: ... 000x 10xx
  else if ((dscb->ds1dsorg & 0x0C) == 0x8)
  {
    *dsorg = ZDS_DSORG_VSAM;
  }

  // Unmovable: Last bit of first half is set
  if ((dscb->ds1dsorg & 0x0100) == 0x0100)
  {
    *dsorg += 'U';
  }
```